catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1452,8 +1452,7 @@ public UpdateResponse update(UpdateRequest updateRequest)
             for (Entry<Serializable, Metacard> update : updateRequest.getUpdates()) {
                 HashMap<String, Set<String>> itemPolicyMap = new HashMap<>();
                 HashMap<String, Set<String>> oldItemPolicyMap = new HashMap<>();
-                Metacard oldMetacard = metacardMap.get(getAttributeStringValue(update.getValue(),
-                        updateRequest.getAttributeName()));
+                Metacard oldMetacard = metacardMap.get(update.getKey());
```

Comment:
Yes. Sounds like that refactor will be in today or tomorrow at which point i will rebase and move this change to the appropriate place.